### PR TITLE
Add SyncReplicaSegments interface for QueryNode

### PR DESCRIPTION
**Pull request by congqixia**
Add SyncReplicaSegments for LoadCollection/LoadPartition child task
This API notifies related ShardCluster to record segment Loaded
Preventing the Etcd events is delayed, which will cause some segment not search

/kind enhancement

Resolves to #16671

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/querynode/shard_cluster.go [hunk 1]
```diff
@@ -243,6 +243,41 @@ func (sc *ShardCluster) updateSegment(evt segmentEvent) {
...
+		sc.segmentCond.L.Unlock()
+	}()
+	sc.mut.Lock()
+	defer sc.mut.Unlock()
+
+	for _, line := range distribution {
+		for _, segmentID := range line.GetSegmentIds() {
+			old, ok := sc.segments[segmentID]
+			if !ok { // newly add
+				sc.segments[segmentID] = &shardSegmentInfo{
```
Comment (yah01) [inline internal/querynode/shard_cluster.go:261]: Should this contain CollectionID field?

File: internal/proto/query_coord.proto [hunk 2]
```diff
@@ -272,6 +273,18 @@ message QueryRequest {
   repeated int64 segmentIDs = 3;
 }
 
+message SyncReplicaSegmentsRequest {
+  common.MsgBase base = 1;
+  string vchannel_name = 2;
```
Comment (yah01) [inline internal/proto/query_coord.proto:278]: Equal to DmChannelName?

Comment (congqixia) [reply internal/proto/query_coord.proto:278]: yes

File: internal/querynode/shard_cluster.go (hunk 1 above)
Comment (congqixia) [reply internal/querynode/shard_cluster.go:261]: vchannel restricts the collection already

**Pull request merged**